template.tpl

```diff
@@ -0,0 +1,2485 @@
+ 
+  const snowplow = copyFromWindow(GLOBALNAME);
+  if (snowplow) {
+    return snowplow;
+  }
+
+  const globalNamespace = createQueue('GlobalSnowplowNamespace');
+  globalNamespace(GLOBALNAME);
+  // Can't use createArgumentsQueue here since the Snowplow tracker library
+  // does not work with GTM's wrapper
+  setInWindow(GLOBALNAME, function() {
+    callInWindow('snowplow.q.push', arguments);
+  });
+  createQueue('snowplow.q');
+  return copyFromWindow(GLOBALNAME);
+};
+const tracker = getSp();
+
+// Helpers
+const fail = msg => {
+  log(ERROR_LOG_PREFIX + msg);
+  return data.gtmOnFailure();
+};
+
+// Helper that returns a valid tracker configuration object
+const getTrackerConfiguration = () => {
+  // Fail if invalid variable
+  if (data.trackerConfigurationVariable !== 'select' && data.trackerConfigurationVariable.type !== 'snowplow') {
+    return false;
+  }
+  // Return if no parameters added
+  if (data.trackerConfigurationVariable === 'select' && (!data.overridingSettings || !data.overridingSettings.length)) {
+    return {};
+  }
+  
+  // Set the configuration to the variable return object
+  const config = data.trackerConfigurationVariable !== 'select' ? data.trackerConfigurationVariable : {};
+  
+  // Override individual fields in the config object from tag settings
+  const overrideSettings = data.overridingSettings && data.overridingSettings.length ? makeTableMap(data.overridingSettings, 'name', 'value') : {};
+  for (let prop in overrideSettings) {
+    config[prop] = overrideSettings[prop];
+  }
+  
+  // Normalize values
+  for (let prop in config) {
+    if (config[prop] === 'null') config[prop] = null;
+    if (config[prop] === 'true') config[prop] = true;
+    if (config[prop] === 'false') config[prop] = false;
+    if (makeNumber(config[prop])) config[prop] = makeNumber(config[prop]);
+  }
+  
+  return config;
+};
+const config = getTrackerConfiguration();
+
+if (!config) {
+  return fail('Could not create a proper tracker configuration object.');
+}
+// Access the generic settings
+const libUrl = (data.spLibrary === 'unpkg' ? UNPKG
+             : (data.spLibrary === 'jsDelivr' ? JSDELIVR
+             : data.selfHostedUrl)) || config.trackerOptions.libUrl;
+const trackerName = data.trackerName || config.trackerOptions.trackerName;
+const endpoint = data.collectorEndpoint || config.trackerOptions.collectorEndpoint;
+
+if (!libUrl) return fail('Missing sp.js library URL');
+if (!trackerName) return fail('Missing tracker name');
+if (!endpoint) return fail('Missing collector endpoint');
+
+// Only initialize the tracker if it hasn't been initialized yet
+if (trackerList.indexOf(trackerName) === -1) {
+  tracker('newTracker', trackerName, endpoint, config);
+  pushToTrackerList(trackerName);
+}
+
+// Load plugins, if any.
+const plugins = data.pluginsTable || [];
+plugins.forEach(plugin => {
+  tracker(
+    'addPlugin',
+    plugin.url,
+    plugin.config.split(','),
+    plugin.additionalConfig || undefined
```

Comment:
A falsey value might be valid here, such as `0` or `false`.

Additionally, if `plugin.additionalConfig` is a string, should we try and convert it into an array (like config with the string split)? The API is always `tracker('addPlugin', string, array, array)`.

e.g.

```
window.snowplow('addPlugin', 
  "https://cdn.jsdelivr.net/npm/@snowplow/browser-plugin-debugger@latest/dist/index.umd.min.js",
  ["snowplowDebugger", "DebuggerPlugin"],
  [ 4 ]
);
```

Although not all the parameters of the array are strings, as shown above, which perhaps makes this challenging to deal with. 

One idea: could we check if its a string, if so, split it on `,` but otherwise just pass the value in (since it could be a JS variable that returns the expected array?). Bit complicated but I dunno of another way.

Reply:
Fixed in e5e97d07272afe863b091cf54cc056d4c3bb8009. Now the log is:

If the input is a string with a comma, split by comma into array.
If the input is an array, keep it as-is.
If the input is something other than an empty string (default if left untouched in GTM UI), undefined, or null, cast into array.
Otherwise set to undefined.